src/key.h

```diff
@@ -203,6 +203,25 @@ class CKey
     ECDHSecret ComputeBIP324ECDHSecret(const EllSwiftPubKey& their_ellswift,
                                        const EllSwiftPubKey& our_ellswift,
                                        bool initiating) const;
+
+    /** Assumes the public key is compressed */
+    template <typename Stream>
+    inline void Unserialize(Stream& s) {
+        std::vector<unsigned char> key_data;
+        key_data.resize(32);
+        s >> MakeWritableByteSpan(key_data);
+        this->Set(key_data.begin(), key_data.end(), true);
+        memory_cleanse(key_data.data(), key_data.size());
+    }
+
+    /** Can only be used for compressed keys, because Unserialize assumes that. */
+    template<typename Stream>
+    void Serialize(Stream &s) const {
+        if (!fCompressed) {
+            throw std::ios_base::failure("Uncompressed key");
```

Comment:
Is it not worth saving `CKey::fCompressed` to disk as well and fully support ser/unser of any `CKey`?